src/components/modals/TransferModal.tsx

```diff
@@ -55,42 +55,56 @@ export const TransferModal = ({
     Airship.clear()
   })
 
-  const handleSend = useHandler(async () => {
-    const result = await Airship.show<WalletListResult>(bridge => (
+  const handleSend = useHandler(() => {
+    Airship.show<WalletListResult>(bridge => (
       <WalletListModal
         bridge={bridge}
         headerTitle={lstrings.select_wallet_to_send_from}
         navigation={navigation}
       />
     ))
-    if (result?.type === 'wallet') {
-      const { walletId, tokenId } = result
-      navigation.push('send2', {
-        walletId,
-        tokenId,
-        hiddenFeaturesMap: { scamWarning: false }
+      .then(result => {
+        if (result?.type === 'wallet') {
+          const { walletId, tokenId } = result
+          navigation.push('send2', {
+            walletId,
+            tokenId,
+            hiddenFeaturesMap: { scamWarning: false }
+          })
+        }
+        Airship.clear()
+      })
+      .catch((error: unknown) => {
+        showError(error)
       })
-    }
-    Airship.clear()
   })
 
-  const handleReceive = useHandler(async () => {
+  const handleReceive = useHandler(() => {
     Airship.clear()
 
-    const result = await Airship.show<WalletListResult>(bridge => (
+    Airship.show<WalletListResult>(bridge => (
       <WalletListModal
         bridge={bridge}
         headerTitle={lstrings.select_receive_asset}
         navigation={navigation}
         showCreateWallet
       />
     ))
-
-    if (result?.type === 'wallet') {
-      const { walletId, tokenId } = result
-      await dispatch(selectWalletToken({ navigation, walletId, tokenId }))
-      navigation.navigate('request', { tokenId, walletId })
-    }
+      .then(result => {
```

Comment:
If you do `.then(async result =>`, then you can keep the rest of the function cleaner. If something goes wrong, the final `.catch` down below will handle that.